app/scripts/app.js

```diff
@@ -378,8 +385,10 @@ angular
     $compileProvider.aHrefSanitizationWhitelist(/^\s*(https?|mailto|git):/i);
   })
   .run(function($rootScope, LabelFilter){
-    $rootScope.$on('$locationChangeSuccess', function(event) {
-      LabelFilter.setLabelSelector(new LabelSelector({}, true), true);
+    // assume we always want filterState persisted, pages that dont can turn it off
+    LabelFilter.persistFilterState(true);
+    $rootScope.$on('$routeChangeSuccess', function() {
+      LabelFilter.readPersistedState();
```

Comment:
realized it made more sense to do it this way, this handles forward/back history correctly which it didnt before